colcon_core/shell/template/prefix.sh.em

```diff
@@ -63,7 +63,23 @@ _colcon_prefix_sh_prepend_unique_value() {
 # add this prefix to the COLCON_PREFIX_PATH
 _colcon_prefix_sh_prepend_unique_value COLCON_PREFIX_PATH "$_colcon_prefix_sh_COLCON_CURRENT_PREFIX"
 unset _colcon_prefix_sh_prepend_unique_value
-@[if pkg_names]@
+
+# use the Python executable known at configure time
+_colcon_python_executable="@(python_executable)"
+# allow overriding it with a custom location
+if [ -n "$COLCON_PYTHON_EXECUTABLE" ]; then
+  _colcon_python_executable="$COLCON_PYTHON_EXECUTABLE"
+fi
+# if the Python executable doesn't exist try another fall back
+if [ ! -f "$_colcon_python_executable" ]; then
+  if /usr/bin/env python3 --version > /dev/null
```

Comment:
has this been tested on osx yet? might have issues based on experience with https://github.com/ros2/ci/pull/127/files

Reply:
Not specifically. It is only meant as a fallback.

Reply:
yeah and the issue I referenced might only be a problem if SIP stripping the DYLD_LIBRARY_PATH is an issue, now that I look closer.. not something that needs to be worried about now